roles/nvidia-gpu-tests/tasks/main.yml

```diff
@@ -0,0 +1,86 @@
+---
+- name: Get a count of GPUs
+  shell: "lspci -d 10de: | grep -c VGA | wc -l"
+  register: gpu_count
+
+# This bit seems a bit buggy depending on the OS 
+- name: Install cuda toolkit
+  become: yes
+  package:
+    name: nvidia-cuda-toolkit
+    state: present
+  when: gpu_test_install_toolkit
+
+- name: download cuda samples
+  git:
+    repo: https://github.com/NVIDIA/cuda-samples.git
+    clone: yes
+    dest: "{{ gpu_test_cuda_dir }}"
+  when: gpu_test_build
+  ignore_errors: yes
+
+- name: make p2p
+  shell: "cd {{ gpu_test_samples_dir }}/p2pBandwidthLatencyTest && make"
+  when: gpu_test_build
+  ignore_errors: yes
+
+- name: make matrixMul
+  shell: "cd {{ gpu_test_samples_dir }}/matrixMul && make"
+  when: gpu_test_build
+  ignore_errors: yes
+
+# Save the output to a log file so we can parse the critical information later
+- name: run p2p test
+  shell: "{{ gpu_test_samples_dir }}/p2pBandwidthLatencyTest/p2pBandwidthLatencyTest | tee {{ gpu_test_samples_dir }}/p2pBandwidthLatencyTest/p2p.log"
+  register: p2p
+  ignore_errors: yes
+
+- name: run matrixmul test
+  shell: "{{ gpu_test_samples_dir }}/matrixMul/matrixMul -wA=4096 -hA=4096 -wB=4096 -hB=4096 | tee {{ gpu_test_samples_dir }}/matrixMul/matrixMul.log"
+  register: matrixmul
+  ignore_errors: yes
+
+# -r 1 for quick system validation test, 2 for mediuem extended validation, 3 for long HW diagnostics/stress
+- name: run dcgm diag for up to 15 minutes
```

Comment:
Is there value in having this vs the other PR (#362)?

Reply:
In this case yes. In the other PR this is being used to collect metrics and health. However, running this command will also stress the system and poke several components that could break.